Okay, so, `raylib-python-cffi` requires `raylib` to be built as a static library on macOS, which succeeds but fails later trying to import symbols from GLFW:
```
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: pyray
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 1, in <lambda>
  File "/nix/store/xnm9sbp02pp2704dvx4kb8g0yxn263pn-python3-3.12.4/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/nix/store/d2jz44lcpi7yzqdilqazkgiqi7qyijkm-python3.12-raylib-python-cffi-5.0.0.2/lib/python3.12/site-packages/pyray/__init__.py", line 18, in <module>
    from raylib import rl, ffi
  File "/nix/store/d2jz44lcpi7yzqdilqazkgiqi7qyijkm-python3.12-raylib-python-cffi-5.0.0.2/lib/python3.12/site-packages/raylib/__init__.py", line 15, in <module>
    from ._raylib_cffi import ffi, lib as rl
ImportError: dlopen(/nix/store/d2jz44lcpi7yzqdilqazkgiqi7qyijkm-python3.12-raylib-python-cffi-5.0.0.2/lib/python3.12/site-packages/raylib/_raylib_cffi.abi3.so, 0x0002): symbol not found in flat namespace '_glfwGetProcAddress'
```

At this point, I went to try and check if the static builds of raylib work on their own and they don't, so yeah, I'm not fixing that

--- a/pkgs/development/python-modules/raylib-python-cffi/default.nix
+++ b/pkgs/development/python-modules/raylib-python-cffi/default.nix
@@ -1,4 +1,6 @@
 {
+  stdenv,
+  gcc,
   buildPythonPackage,
   fetchFromGitHub,
   setuptools,
@@ -9,9 +11,19 @@
   raylib,
   physac,
   raygui,
-  lib
+  darwin,
+  lib,
 }:
 
+let
+  inherit (darwin.apple_sdk.frameworks)
+    OpenGL
+    Cocoa
+    IOKit
+    CoreFoundation
+    CoreVideo
+    ;
+in
 buildPythonPackage rec {
   pname = "raylib-python-cffi";
   version = "5.0.0.2";
@@ -31,22 +43,36 @@ buildPythonPackage rec {
     # This patch fixes to the builder script function to call pkg-config
     # using the library name rather than searching only through raylib
     ./fix_pyray_builder.patch
+
+    # use get_lib_flags() instead of linking to libraylib.a directly
+    ./fix_macos_raylib.patch
   ];
 
-  nativeBuildInputs = [ pkg-config ];
+  nativeBuildInputs = [
+    pkg-config
+    gcc
+  ];
 
   # tests require a graphic environment
   doCheck = false;
 
   pythonImportsCheck = [ "pyray" ];
 
-  buildInputs = [
-    glfw
-    libffi
-    raylib
-    physac
-    raygui
-  ];
+  buildInputs =
+    [
+      glfw
+      libffi
+      raylib
+      physac
+      raygui
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      OpenGL
+      Cocoa
+      IOKit
+      CoreFoundation
+      CoreVideo
+    ];
 
   meta = {
     description = "Python CFFI bindings for Raylib";

--- a/pkgs/development/python-modules/raylib-python-cffi/fix_macos_raylib.patch
+++ b/pkgs/development/python-modules/raylib-python-cffi/fix_macos_raylib.patch
@@ -1,0 +1,11 @@
+--- a/raylib/build.py
++++ b/raylib/build.py
+@@ -158,7 +158,7 @@ def build_unix():
+
+     if platform.system() == "Darwin":
+         print("BUILDING FOR MAC")
+-        extra_link_args = [get_the_lib_path() + '/libraylib.a', '-framework', 'OpenGL', '-framework', 'Cocoa',
++        extra_link_args = get_lib_flags() + ['-framework', 'OpenGL', '-framework', 'Cocoa',
+                            '-framework', 'IOKit', '-framework', 'CoreFoundation', '-framework',
+                            'CoreVideo']
+         libraries = []

